packages/cactus-cmd-api-server/src/test/typescript/integration/jwt-unprotected-endpoint-authz.test.ts

```diff
@@ -26,102 +26,94 @@ const log = LoggerProvider.getOrCreate({
+  let apiServer: ApiServer;
+
+  afterAll(async () => {
+    await apiServer.shutdown();
+  });
+
+  test(testCase, async () => {
+    try {
+      const jwtKeyPair = await generateKeyPair("RS256", {
+        modulusLength: 4096,
+      });
+      const jwtPublicKey = await exportSPKI(jwtKeyPair.publicKey);
+      const expressJwtOptions: expressJwt.Options = {
+        algorithms: ["RS256"],
+        secret: jwtPublicKey,
+        audience: uuidv4(),
+        issuer: uuidv4(),
+      };
+      expect(expressJwtOptions).toBeTruthy();
+
+      const ep = new UnprotectedActionEndpoint({
+        connector: {} as PluginLedgerConnectorStub,
+        logLevel,
+      });
+
+      const authorizationConfig: IAuthorizationConfig = {
+        unprotectedEndpointExemptions: [ep.getPath()],
+        expressJwtOptions,
+        socketIoJwtOptions: { secret: jwtPublicKey },
+      };
+
+      const pluginRegistry = new PluginRegistry();
+      const plugin = new PluginLedgerConnectorStub({
+        logLevel,
+        pluginRegistry,
+        instanceId: uuidv4(),
+      });
+      pluginRegistry.add(plugin);
+
+      const configService = new ConfigService();
+      const apiSrvOpts = await configService.newExampleConfig();
+      apiSrvOpts.authorizationProtocol = AuthorizationProtocol.JSON_WEB_TOKEN;
+      apiSrvOpts.authorizationConfigJson = authorizationConfig;
+      apiSrvOpts.configFile = "";
+      apiSrvOpts.apiCorsDomainCsv = "*";
+      apiSrvOpts.apiPort = 0;
+      apiSrvOpts.cockpitPort = 0;
+      apiSrvOpts.grpcPort = 0;
+      apiSrvOpts.apiTlsEnabled = false;
+      apiSrvOpts.plugins = [];
+      const config = await configService.newExampleConfigConvict(apiSrvOpts);
+
+      apiServer = new ApiServer({
+        config: config.getProperties(),
+        pluginRegistry,
+      });
+
+      const startResponse = apiServer.start();
+      expect(startResponse).not.toReject();
+      expect(startResponse).toBeTruthy();
+
+      const addressInfoApi = (await startResponse).addressInfoApi;
+      const protocol = apiSrvOpts.apiTlsEnabled ? "https" : "http";
+      const { address, port } = addressInfoApi;
+      const apiHost = `${protocol}://${address}:${port}`;
+
+      const req1 = {
+        requestId: uuidv4(),
+      };
+
+      // look Ma, no access token
+      const res1 = await axios.request({
+        data: req1,
+        url: `${apiHost}${ep.getPath()}`,
+        method: ep.getVerbLowerCase() as Method,
+      });
+      expect(res1).toBeTruthy();
+      expect(res1.status).toBe(StatusCodes.OK);
+      expect(typeof res1.data).toBe("object");
+      expect(typeof res1.data.data).toBeTruthy();
+      expect(typeof res1.data.data.reqBodyrequestId).toBeTruthy();
+      expect(typeof res1.data.data.reqBody.requestId).toBeTruthy();
+      expect(res1.data.data.reqBody.requestId).toBe(req1.requestId);
+    } catch (ex) {
+      log.error(ex);
+      // t.fail("Exception thrown during test execution, see above for details!");
```

Comment:
@Leeyoungone There is a way in Jest to fail the test explicitly as well. Please use that so that whoever is debugging a failing test later will have a little more info.

Reply:
ah sorry, I had just forgotten to jestify that part. Good catch :)